src/SOS/Strike/CMakeLists.txt

```diff
@@ -155,30 +173,19 @@ else(WIN32)
 
   if(CMAKE_SYSTEM_NAME STREQUAL Linux OR CMAKE_SYSTEM_NAME STREQUAL FreeBSD OR CMAKE_SYSTEM_NAME STREQUAL NetBSD)
       set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Xlinker -Bsymbolic -Xlinker -Bsymbolic-functions")
-
-      # These options are used to force every object to be included even if it's unused.
-      set(START_WHOLE_ARCHIVE -Wl,--whole-archive)
-      set(END_WHOLE_ARCHIVE -Wl,--no-whole-archive)
-
       set(EXPORTS_LINKER_OPTION -Wl,--version-script=${EXPORTS_FILE})
   endif(CMAKE_SYSTEM_NAME STREQUAL Linux OR CMAKE_SYSTEM_NAME STREQUAL FreeBSD OR CMAKE_SYSTEM_NAME STREQUAL NetBSD)
 
   if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
-      # These options are used to force every object to be included even if it's unused.
-      set(START_WHOLE_ARCHIVE -force_load)
-      set(END_WHOLE_ARCHIVE )
-
       set(EXPORTS_LINKER_OPTION -Wl,-exported_symbols_list,${EXPORTS_FILE})
   endif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
 
   set(SOS_LIBRARY
     extensions
     corguids
     dbgutil
-    ${START_WHOLE_ARCHIVE} # force all PAL objects to be included so all exports are available
-    coreclrpal
```

Comment:
We no longer need to root symbols here?

Reply:
I think that was previously a mistake and doesn't allow the linker to remove unused code/functions in the pal.  We used it in the DAC in the runtime so all the PAL exports are not optimized away. This should make libsos smaller.

Reply:
Ah right, SOS doesn't need to preserve like the DAC does 👍🏼